bundle/src/main/java/com/adobe/acs/commons/util/impl/RequireAemImpl.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.util.impl;
+
+import com.adobe.acs.commons.util.RequireAem;
+import com.adobe.granite.license.ProductInfo;
+import com.adobe.granite.license.ProductInfoProvider;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.framework.Version;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+/**
+ * ACS AEM Commons - RequireAem
+ *
+ * Allows ACS Commons features determine if they are running on AEM as a Cloud Service or not.
+ */
+@Component(
+        service = {},
+        immediate = true
+)
+public class RequireAemImpl implements RequireAem {
+    private static final Logger log = LoggerFactory.getLogger(RequireAemImpl.class);
+    private static final String PN_CLOUD_READY = "cloud-ready";
```

Comment:
what about `distribution` with values `cloud-ready` and `classic` or `onprem`? That way we could extend that further, like `cloud-ready-minimum`, `cloud-ready-forms` and whatever may come up in the future. Maybe just as addition, but I really see that in the future we may have even more than those two distributions...

Reply:
that works... im leaning towards `classic` since AMS is typically not considered "onprem" ..

Reply:
@kwin Ok - updated the Impl per your suggestion